src/api/user/user.ts

```diff
@@ -30,32 +30,18 @@ router.post(
   }),
 )
 
-router.post(
-  '/edit',
+router.patch(
+  '/',
   validationMiddleware(userEditSchema),
-  asyncHandler(async (req, res) => {
-    const user = await UserModel.getByEmail(req.body.email)
-    const data = {
-      name: req.body.name,
-      email: req.body.email,
-      password: req.body.password,
-    }
-    if (!user) {
-      throw new BadRequestError({
-        error: 'User not found.',
-      })
-    }
-    if (req.body.password !== user.password) {
-      throw new BadRequestError({
-        error: 'Your credentials are invalid',
-      })
-    }
-    const updatedUser = await UserModel.update(user.userId, data)
-    if (user) {
-      res.send('User updated successfully')
-    } else {
+  verifyToken,
+  asyncHandler(async (req: any, res) => {
+    const user = req.authData.user
+    const data = req.body
+    const updatedUser = await UserModel.update(user.user_id, data)
```

Comment:
If the error happens, the server will return error 500 instead of badrequest.
Mainly because you are throwing a normal Error inside the model and not handling it here